ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx

```diff
@@ -94,18 +141,29 @@ const ProfileIcicleGraph = ({
       return;
     }
     setActionButtons(
-      <>
-        <Button
-          color="neutral"
-          onClick={() => setNewCurPath([])}
-          disabled={curPath.length === 0}
-          variant="neutral"
-        >
-          Reset View
-        </Button>
-      </>
+      <div className="flex w-full justify-end gap-2 pb-2">
+        <div className="flex w-full items-center justify-between space-x-2">
+          {table !== undefined && (
+            <>
+              <GroupByDropdown groupBy={groupBy} toggleGroupBy={toggleGroupBy} />
+              <SortBySelect compareMode={compareMode} sortBy={localSortBy} setSortBy={setSortBy} />
+            </>
+          )}
+          <div>
+            <label>&nbsp;</label>
+            <Button
+              color="neutral"
+              onClick={() => setNewCurPath([])}
+              disabled={curPath.length === 0}
+              variant="neutral"
+            >
+              Reset View
+            </Button>
+          </div>
+        </div>
+      </div>
     );
-  }, [setNewCurPath, curPath, setActionButtons]);
+  }, [setNewCurPath, curPath, setActionButtons, table, compareMode, localSortBy, groupBy]);
```

Comment:
Adding `setSortBy` or `toggleGroupBy` makes the component re-render endlessly. Both of those callbacks are wrapped with `useCallback`.